Base/res/html/directory.html

```diff
@@ -0,0 +1,54 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <meta charset="UTF-8">
+        <title>Index of @path@</title>
+        <style>
+            header {
+                margin-bottom: 10px;
+            }
+            h1 {
+                display: inline;
+            }
+            a {
+                text-decoration: none;
+            }
+            a:focus, a:hover {
+                text-decoration: underline;
+            }
+            table {
+                font-family: monospace;
+            }
+
+            .folder, .file, .open-parent {
+                display: inline-block;
+                width: 16px;
+                height: 16px;
+                margin-right: 5px;
+                background-size: contain;
+            }
+            .folder {
+                /* /res/icons/32x32/filetype-folder.png */
+                background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAC1ElEQVRYw+2XMW4VQRBE36If+gaIADkkICBDBD8gICXiBJzDk3EiQqSNOASShQQEiMARpF0EOzNdvbYFWAYCWGnl2f2zPd1V1TVj+NevZf+itaYffdRaW35LNq01/eha11U/k+QvI9Ba09nZGZ9ev+L8/D2np/ev/3vymHVdbwWJO7+8+Pl7Tr++5Xg8chtILK01HY/HGwdY1/WmdAMsB4BH906YpUioPwgb94GwZ8GDF8+2mfOH/t2cnGON+MDL50+5+/CJDmPdb+/eEBIK9YX7HRCKLVCIEIiAPkaBJKInJCLjjBhscaSY449fLgA4WOHb5AUoHzMr3MbR0djuMFiDGIEqKsEsCm2JDCgPA6Qg+nejsqRgJqcwSpIGSvAxt4PfCxkcxExZhsAsSER/0CBNHiTnzQV9kWsWn5SSSUZFgPwIJvwhSwYRMYB2eLekNcvvDJiOXLSyZCsCg4IRfKBmd/mNIdiN400PqZmwgiI0u0uG9tUilDJY5zw5VEVpcYj39GTFybKsTVVFmMF7sEiBAdlWgCKstWRwU2CW9uKswi4IBA5T5pjPaSRdzr2vKWblQvOEcBS1QwAPwk447D50lbPvEs1Oko2TUherU+DQTWFVFxsWWvrc4cc4H1owNFzNbvGGgPd6WIWTIEp7TEG5U4bxbG1H7BKfrTMQEOYCZBEq1WZXgLRM32eI1gS7gZhQTx1d6QN98lC1sE2JbCsZtJK/H++WNC8zsUtUJaaZADLo3WzKdqxrXW6vCXZaSW30eXFJA5H79mgvsAqHh8sOBcOYbfHuH1FauhqdlHvowc4N5r2UzUbucBiXhpaoO6NcS+WMQTnYHEYvB8bvLNBUjesg6iKbeC7thPh5IjLqaPcdAsE4JxQ/UORms3PFTCaPXK5w7yo3OTeyuRt++HxRxdS342oy0ZMwFMzdYiQaKtpBUTVhCSxUSv76f2b/rz9+fQe7BdtNIJY3gAAAAABJRU5ErkJggg==');
```

Comment:
Why do we need to embed the PNG as a data url? Do any of our other default pages do that?

Reply:
The current `error.html` page template that is loaded in `FrameLoader::load_error_page()` lies over it's own URL / path so that it can load relative icons that are in Base/res/. I can't find a way to do that the same way in `FileDirectoryLoader.cpp`.

But may be I should change it to the way the icons in `Services/WebServer/Client.cpp:230` works. The PNG file is loaded from the `Base/res/` folder then encoded to base64 that string is send to the template file. That way we have less duplication.

Reply:
I have taken a even easier choice, I add the `FrameLoader::set_resource_directory_url()` API where `Ladybird/main.cpp` can set the relaive `file://` path URL to the `Base/res`. That variable is used in the internal HTML pages (`error.html`, `directory.html`) to point to the right image files. This is much better then embedding base64 PNG images that can get outdated.